### PR TITLE
Add shared memory

### PR DESCRIPTION
**Pull request by swolferDF**
add shem to pyopendds 
can cause problem when we create two domain with same int one in rtps and other with shem in the same code

### CODE REVIEW

File: pyopendds/DomainParticipant.py [hunk 1]
```diff
@@ -13,18 +13,16 @@ def participant_cleanup(*args):
...
         self.listener = listener
         self.topics = {}
         self.subscribers = []
         self.publishers = []
         self._registered_typesupport = []
-
         from _pyopendds import create_participant  # noqa
-        create_participant(self, domain)
-
+        create_participant(self, domain,int(isRtpstransport))
```
Comment (Sdpierret) [inline pyopendds/DomainParticipant.py:25]: You don't need to explicit tye as int
Booleans are subclass of Integers
```
>>> isinstance(True, int)
True
>>> issubclass(bool, int)
True
```

File: pyopendds/ext/_pyopendds.cpp [hunk 2]
```diff
@@ -210,16 +203,40 @@ void delete_participant_var(PyObject* part_capsule)
 PyObject* create_participant(PyObject* self, PyObject* args)
 {
     Ref pyparticipant;
-    unsigned domain;
-    if (!PyArg_ParseTuple(args, "OI", &*pyparticipant, &domain)) {
+    unsigned int domain;
+    int isRtpstransport;
+    if (!PyArg_ParseTuple(args, "OIi", &*pyparticipant, &domain,&isRtpstransport)) {
```
Comment (Sdpierret) [inline pyopendds/ext/_pyopendds.cpp:208]: pep : space after comma

File: pyopendds/ext/_pyopendds.cpp [hunk 3]
```diff
@@ -210,16 +203,40 @@ void delete_participant_var(PyObject* part_capsule)
...
+        }
+    }
+    transport_config->instances_.push_back(transport_inst);
+    TheTransportRegistry->domain_default_config(domain, transport_config);
+   
     // Create Participant
     DDS::DomainParticipantQos qos;
     participant_factory->get_default_participant_qos(qos);
-
+
```
Comment (Sdpierret) [inline pyopendds/ext/_pyopendds.cpp:239]: trailing spaces

**Review by Sdpierret — COMMENTED** [review]
why moving the transport on the create participant ?
we can have 2 different transport in the same instance ?

**Comment by swolferDF** [comment]
we can have 2 different transport in the same instance ? 
-> yes if we have 2 domain in the same instance

**Review by Sdpierret — APPROVED** [review]

**Pull request merged**